pandas/core/arrays/interval.py

```diff
@@ -543,33 +543,81 @@ def __eq__(self, other):
             # extract intervals if we have interval categories with matching closed
             if is_interval_dtype(other_dtype):
                 if self.closed != other.categories.closed:
-                    return np.zeros(len(self), dtype=bool)
+                    return invalid_comparison(self, other, op)
                 other = other.categories.take(other.codes)
 
         # interval-like -> need same closed and matching endpoints
         if is_interval_dtype(other_dtype):
             if self.closed != other.closed:
-                return np.zeros(len(self), dtype=bool)
-            return (self._left == other.left) & (self._right == other.right)
+                return invalid_comparison(self, other, op)
+            if isinstance(other, Interval):
+                other = type(self)._from_sequence([other])
+                if self._left.dtype.kind in ["m", "M"]:
+                    # Need to repeat bc we do not broadcast length-1
+                    # TODO: would be helpful to have a tile method to do
+                    #  this without copies
+                    other = other.repeat(len(self))
+            else:
+                other = type(self)(other)
+
+            if op is operator.eq:
+                return (self._left == other._left) & (self._right == other._right)
+            elif op is operator.ne:
+                return (self._left != other._left) | (self._right != other._right)
+            elif op is operator.gt:
+                return (self._left > other._left) | (
+                    (self._left == other._left) & (self._right > other._right)
+                )
+            elif op is operator.ge:
+                return (self == other) | (self > other)
+            elif op is operator.lt:
+                return (self._left < other._left) | (
+                    (self._left == other._left) & (self._right < other._right)
+                )
+            else:
+                # operator.lt
+                return (self == other) | (self < other)
 
         # non-interval/non-object dtype -> no matches
         if not is_object_dtype(other_dtype):
-            return np.zeros(len(self), dtype=bool)
+            return invalid_comparison(self, other, op)
 
         # object dtype -> iteratively check for intervals
         result = np.zeros(len(self), dtype=bool)
-        for i, obj in enumerate(other):
-            # need object to be an Interval with same closed and endpoints
-            if (
-                isinstance(obj, Interval)
-                and self.closed == obj.closed
-                and self._left[i] == obj.left
-                and self._right[i] == obj.right
-            ):
-                result[i] = True
-
+        try:
+            for i, obj in enumerate(other):
+                result[i] = op(self[i], obj)
+        except TypeError:
+            # pd.NA
```

Comment:
Do you have an example (or test) that runs into this?

Reply:
Ah, I see the removed special case below in the tests. Now, this is certainly in a messy state, but I am not sure the conversion to object dtype is needed here. Currently, we still return False (and not NA as for nullable dtypes) on comparisons with NA:

```
In [90]: arr = pd.interval_range(0,3).array

In [91]: arr == pd.NA
Out[91]: array([False, False, False])

In [92]: arr[0] == pd.NA
Out[92]: False
```